graph/registry.go

```diff
@@ -91,7 +91,7 @@ func NewV2Repository(repoInfo *registry.RepositoryInfo, endpoint registry.APIEnd
 	}
 
 	creds := dumbCredentialStore{auth: authConfig}
-	tokenHandler := auth.NewTokenHandler(authTransport, creds, repoName, "push", "pull")
+	tokenHandler := auth.NewTokenHandler(authTransport, creds, repoName, actions)
```

Comment:
This can be `tokenHandler := auth.NewTokenHandler(authTransport, creds, repoName, actions...)`. No need to change function signature.